File: balancedBraces.js
const s = `function isBalanced(s, braces) {
            if(!s.length) {
            }
        }`;

const definedOpenBraces="{[(<";
const definedCloseBraces="}])>";

let braces = [];
let stage = 0;

function isBalanced(s, braces) {
    filterString = definedCloseBraces + definedOpenBraces;
    if(!s.length)
        return;
    let firstOccurance = 0;
    for(let i=0; i < s.length; i++) {
        if(definedOpenBraces.indexOf(s[i]) > -1) {
            firstOccurance = i;
            braces.push(s[i]);
            stage++;
            console.log(stage, braces);
            break;
        } else if (definedCloseBraces.indexOf(s[i]) > -1) {
            firstOccurance = i;
            index = definedCloseBraces.indexOf(s[i]);
            if(braces[braces.length-1] === definedOpenBraces[index]) {
                braces.pop();
                stage--;
                console.log(stage, braces);
                break;
            } else{
                console.log(braces, "not bananced");
                return;
            }
        }
    }
    isBalanced(s.substr(firstOccurance + 1), braces);
}
isBalanced(s, braces);
console.log("string is ", braces.length ? "un-Balanced" : "balanced");

File: balancedBraces.cpp
#include <iostream>
#include <stack>
using namespace std;

const string definedOpenBraces = "{[(<";
const string definedloseBraces = "}])>";
enum braceType { open, none, close };
void isBalanced(string s, stack<char>& braces);

int main()
{
	string s = "[()]<>{}()";
	stack<char> braces;
	isBalanced(s, braces);
	string result = braces.size() == 0 ? "\tYes, string is balanced" : "\tNo, string is un-balanced";
	cout << result << endl;
}

braceType stringContainsBraces(char ch)
{
	braceType result = none;
	for (int i = 0; i < definedOpenBraces.size(); i++) {
		if (ch == definedOpenBraces[i])
			result = open;
		if (ch == definedloseBraces[i])
			result = close;
	}
	return result;
}
void isBalanced(string s, stack<char> & braces)
{
	if (s.length() == 0)
		return;
	int firstOccurance = 0;
	for (int i = 0; i < s.length(); i++) {
		if (stringContainsBraces(s[i]) == open) {
			firstOccurance = i;
			braces.push(s[i]);
			cout << '\t' << s[i] << endl;
			break;
		}
		else if (stringContainsBraces(s[i]) == close) {
			int index = definedloseBraces.find(s[i]);
			if (braces.top() == definedOpenBraces[index]) {
				braces.pop();
				cout << '\t' << s[i] << endl;
				break;
			}
			else {
				cout << "\terror:\t" << braces.top() << " != " << s[i] << endl;
				return;
			}
		}
	}
	isBalanced(s.substr(firstOccurance + 1), braces);
}